custom-instrumentation/instrument/instrumentation.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const newrelic = require('newrelic')
+const queuePath = require.resolve('./job-queue')
+
+newrelic.instrument({
+    // The absolute path to the required module
+    absolutePath: queuePath, 
+    // The module's name 
+    moduleName: 'job-queue',
+    // The function that will be called once the module is required
+    onRequire: function onRequire(shim, jobQueue) {
+        console.log(`[NEWRELIC] instrumenting job-queue module`)
+
+        console.log(`[NEWRELIC] instrumenting method 'scheduleJob'`)
+        shim.record(
+            jobQueue.prototype,
+            'scheduleJob',
+            function wrapJob(shim, original) {
```

Comment:
this is recording segments but not naming them or anything.  This is more of a `shim.wrap` example, same for the one below

Reply:
![Screenshot 2024-07-31 at 1 15 25 PM](https://github.com/user-attachments/assets/287f214b-24e4-4b81-b432-b44664a0f45d)
It shows up in NR like this, isn't this essentially naming it?

Reply:
yes that's a use case for binding a segment. but you don't need `shim.record` to do that. you can just use `shim.wrap`. shim.record expects you to return a spec object that has naming and other things. just take a look at the examples in custom-instrumetation/instrument-messages as an example or instrument-webframework.